src/solver/optimisation/adequacy_patch_csr/csr_quadratic_problem.cpp

```diff
@@ -104,161 +74,47 @@ void CsrQuadraticProblem::setNodeBalanceConstraints(std::vector<double>& Pi, std
-        while (Interco >= 0)
-        {
-            if (problemeHebdo_->adequacyPatchRuntimeData->originAreaMode[Interco]
-                != Data::AdequacyPatch::physicalAreaInsideAdqPatch)
-            {
-                Interco = problemeHebdo_->IndexSuivantIntercoExtremite[Interco];
-                continue;
-            }
-            var = CorrespondanceVarNativesVarOptim
-                    .NumeroDeVariableDeLInterconnexion[Interco]; // flow (2 -> A)
-            if (var >= 0)
-            {
-                Pi[NombreDeTermes] = 1.0;
-                Colonne[NombreDeTermes] = var;
-                NombreDeTermes++;
-                logs.debug()
-                  << "E-Interco number: [" << std::to_string(Interco) << "] between: ["
-                  << problemeHebdo_->NomsDesPays[Area] << "]-["
-                  << problemeHebdo_
-                       ->NomsDesPays[problemeHebdo_->PaysOrigineDeLInterconnexion[Interco]]
-                  << "]";
-            }
-            Interco = problemeHebdo_->IndexSuivantIntercoExtremite[Interco];
-        }
-
-        // - Spilled Energy
-        var = CorrespondanceVarNativesVarOptim.NumeroDeVariableDefaillanceNegative[Area];
-        if (var >= 0)
-        {
-            Pi[NombreDeTermes] = -1.0;
-            Colonne[NombreDeTermes] = var;
-            NombreDeTermes++;
-        }
-
-        hourlyCsrProblem_.numberOfConstraintCsrAreaBalance[Area]
-          = problemeAResoudre_.NombreDeContraintes;
-
-        std::string NomDeLaContrainte
-          = "Area Balance, Area:" + std::to_string(Area) + "; " + problemeHebdo_->NomsDesPays[Area];
-
-        logs.debug() << "C: " << problemeAResoudre_.NombreDeContraintes << ": "
-                     << NomDeLaContrainte;
-
-        OPT_ChargerLaContrainteDansLaMatriceDesContraintes(
-          &problemeAResoudre_, Pi, Colonne, NombreDeTermes, '=');
-    }
+    auto builder = NewGetConstraintBuilderFromProblemHebdoAndProblemAResoudre(problemeHebdo_,
+                                                                              problemeAResoudre_);
+    CsrAreaBalance csrAreaBalance(builder);
+
+    CsrAreaBalanceData data{problemeHebdo_->adequacyPatchRuntimeData->areaMode,
+                            hour,
+                            problemeHebdo_->IndexDebutIntercoOrigine,
+                            problemeHebdo_->IndexSuivantIntercoOrigine,
+                            problemeHebdo_->IndexDebutIntercoExtremite,
+                            problemeHebdo_->IndexSuivantIntercoExtremite,
+                            problemeHebdo_->adequacyPatchRuntimeData->originAreaMode,
+                            problemeHebdo_->adequacyPatchRuntimeData->extremityAreaMode,
+                            problemeHebdo_->PaysOrigineDeLInterconnexion,
+                            problemeHebdo_->PaysExtremiteDeLInterconnexion,
+                            hourlyCsrProblem_.numberOfConstraintCsrAreaBalance,
```

Comment:
This is dangerous. Two members could easily be swapped mistakenly. This kind of error is very hard to spot

Use the `{.a=a, .b=b}` syntax ?